File: 7kyu/FixStringCase/fix_string_case.go
package main

import (
	"fmt"
	"strings"
)

// fixStringCase returns a string with the same case. The case with the large number is adopted by the whole string. Defaults to lowercase.
func fixStringCase(str string) string {
	uppercaseCount := 0
	lowercaseCount := 0

	for _, s := range str {
		if string(s) == strings.ToUpper(string(s)) {
			uppercaseCount++
		} else if strings.ToLower(string(s)) == string(s) {
			lowercaseCount++
		}
	}

	if uppercaseCount > lowercaseCount {
		return strings.ToUpper(str)
	}

	return strings.ToLower(str)
}

func main() {
	fmt.Println(fixStringCase("coDe"))
	fmt.Println(fixStringCase("CODe"))
	fmt.Println(fixStringCase("COde"))
}

File: 7kyu/FixStringCase/fix_string_case.js
const fixStringCase = (str) => {
  let upperCase = 0
  let lowerCase = 0
  str.split('').forEach(e => e.toUpperCase() === e ? upperCase ++ : lowerCase++)
  
  return upperCase > lowerCase ? str.toUpperCase() : str.toLowerCase()
}

console.log(fixStringCase('coDe'))
console.log(fixStringCase('CODe'))
console.log(fixStringCase('COde'))